### PR TITLE
Add(route): add Codeforces contests

### LINKED ISSUE #8814: 希望添加 codeforces 比赛列表
### 类型

编程

### 网站地址

https://codeforces.com/contests

### 网站描述

希望添加 codeforces 比赛列表

### 需要生成什么内容？

codeforces 比赛列表

### 额外描述

_No response_

### 这不是重复的 RSS 请求

- [X] 我已经搜索了[现有 issue](https://github.com/DIYgod/RSSHub/issues) 和 [pull requests](https://github.com/DIYgod/RSSHub/pulls)，以确保该 RSS 尚未被请求。

### PR DESCRIPTION
**Pull request by Fatpandac**
## 该 PR 相关 Issue / Involved issue

Close #8814

## 完整路由地址 / Example for the proposed route(s)

```routes
/codeforces/contests
```

## 新RSS检查列表 / New RSS Script Checklist
  
- [X] New Route
- [X] Documentation
  - [X] CN
  - [X] EN
- [ ] 全文获取 fulltext
  - [ ] Use Cache
- [ ] 反爬/频率限制 anti-bot or rate limit?
  - [ ] 如果有, 是否有对应的措施? If yes, do your code reflect this sign?
- [X] 日期和时间 date and time
  - [X] 可以解析 Parsed
  - [X] 时区调整 Correct TimeZone
- [ ] 添加了新的包 New package added 
- [ ] `Puppeteer`

### CODE REVIEW

**Review by TonyRL — CHANGES_REQUESTED** [review]
Please provide en docs for this route.

**Comment by Fatpandac** [comment]
> Please provide en docs for this route.



> Please provide en docs for this route.

Done.

File: lib/v2/codeforces/contests.js [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+                startTime: timezone(parseDate(contest.startTimeSeconds * 1000), +0),
+                durationTime: contest.durationSeconds / 60,
+                relativeTime: Math.floor(contest.relativeTimeSeconds / 60),
+                type: contest.type,
+            });
+
+            return {
+                title,
+                description,
+                link: 'https://codeforces.com/contests',
```
Comment (a180285) [inline lib/v2/codeforces/contests.js:28]: Thank you very much for help me for this Route Request. one suggestion is that if the link for each item can navigate me to the direct contest page. That will be much helpful.

```suggestion
                link: 'https://codeforces.com/contests/' + contest.id,
```

Comment (Fatpandac) [reply lib/v2/codeforces/contests.js:28]: OK

**Comment by a180285** [comment]
LGTM

**Pull request merged**